langkit/expressions/collections.py

```diff
@@ -778,6 +778,30 @@ def collection_get(self, collection, index, or_null):
     return result
 
 
+@auto_attr
```

Comment:
I'm not 100% sure that it is possible (only 95%), but I think this should be a builtin property: You'll vastly decrease the amount of code necessary, it will be much easier to port to the new DSL (basically nothing to do), and you'll not block `to_iterator` for other uses.

It might also be much easier to expand to (for example) AST lists if needed.

Reply:
:disappointed: I'm so sorry